### ISSUE TITLE
Frameless window size on a second display

### ISSUE THREAD

**Issue by me-vlad** [rse/vingester#35]
When using second display for frameless fullscreen output vingester creates 1536x834 window on a fullhd (1920x1080) display.
Same issue when I used Blackmagic ATEM switcher as a second display.
Vingester version:  2.2.4
OS: Windows 10 Pro x86_64 21H1
Intel HD Graphics 4600 (driver version 20.19.15.5171)
Nvidia GeForce GTX 860M  (driver version 27.21.14.6663)

P.S. On a Fedora 34 dual display setup works as expected.

Screenshot:

![image](https://user-images.githubusercontent.com/222185/121326216-03cd0580-c91b-11eb-9646-d173abce466f.png)

**Comment by rse** [comment]
Vingestern uses the DOTs and not PIXELs when displaying the frameless windows. Your display seems to use a 125% display scaling. The 1536x834 PIXELs are actually 1920x1080 DOTs and hence it IS the full resolution your feed it. Just try to capture the window and it should be 1080p in size.

**Comment by me-vlad** [comment]
Thanks for explanation, Ralf!
I this case I see another issue - Vingester detects display scaling on the primary display where Vingester main app window started, not the secondary display scaling where placed a frameless browser window. On my screenshot secondary display configured to 100% scale.
Shure, I can start Vingester on another display (not in case when video mixer used as a "display") or change scaling on a primary display to 100% as a workaround. It will be cool to autodetect display scaling for frameless window, not for main application window, if  possible.

P.S. Not relative to this issue, just want to say that your app perfectly fits in my workflow, thanks!

**Comment by rse** [comment]
I've checked the Vingester code and according to it, the scale factor of the particular selected display is actually used. And I've tested Vingester 2.2.5 now with two identical 4K displays: the primary one is 150% scale factor, the other 100% scale factor. And on both displays the opened Vingester frameless windows are exactly the same physical size -- so the scale factor of the particular display is correctly applied. Can you recheck this, please? Perhaps in your various tests you mixed up something?

**Comment by me-vlad** [comment]
Just one more thing about display detection.
If I have not enable Windows toolbar autohide feature before starting frameless output - browser window height is smaller on a toolbar size even if "Pin" mode enabled and toobar turned off (using Windows Settings) on a secondary display.

Screenshot:

![image](https://user-images.githubusercontent.com/222185/121384213-bb303f00-c950-11eb-9424-661cd65fe084.png)

**Comment by rse** [comment]
This problem with the toolbar of Windows looks like a bug. I would have expected that at least with "Pin" enabled, the toolbar does not matter. I've to check this...

**Comment by me-vlad** [comment]
Just upgraded Vingester to 2.2.5 and checked in different scaling modes with and w/o toolbar autohide.
My results:
| Displays scaling settings | Frameless window size without toolbar autohide | Frameless window size with toolbar autohide |
|:---|:---|:---|
**Primary** display with **higher** scale (125% vs 100%) | 1536x834 | 1536x834 |
**Equal** scaling (125% vs 125% or 100% vs 100%) | 1920x1045 | 1920x 1083 |
**Primary** display with **lower** scale (125% vs 150%) | 1923x1083 | 1923x1080 |

**Comment by rse** [comment]
Ok, in Vingester 2.2.6 I've now tried to apply some band-aid for forcing the correct frameless window size. At least under Windows 10 with the taskbar in auto-hide and normal mode, the windows were opened correctly now. At least https://vingester.app/test/ (with the Sample-TEST.yaml) indicates the correct sizes. Can you retest with 2.2.6?

**Comment by me-vlad** [comment]
Version 2.2.6 on Win10 works fine - frameless window size on a secondary display exactly 1920x1080 without tollbar workarounds even with enabled toolbar on all displays and Pin mode turned off.

**Comment by rse** [comment]
Great. Thanks for re-testing. Good to know that 2.2.6 now works fine.
